src/containers/EnterBarcodePage.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { useState, useEffect } from "react";
+import "../styles/HomeLandingPage.css";
+import Typography from "@material-ui/core/Typography";
+import Button from "@material-ui/core/Button";
+import FormField from "../components/common/FormField";
+import { NavLink } from "react-router-dom";
+import CompleteBarcodeButton from "../components/EnterBarcodePage/CompleteBarcodeButton";
+import { useHistory } from "react-router-dom";
+import { useQuery } from "react-apollo";
+import { FETCH_ALL_PATIENTS } from "../graphql/queries/patients";
+
+const EnterBarcodePage = ({
+  match: {
+    params: { ccpID },
+  },
+}: {
+  match: { params: { ccpID: string } };
+}) => {
+  const history = useHistory();
+  const { pathname } = history.location;
+  const { data, loading } = useQuery(FETCH_ALL_PATIENTS);
+  const [barcode, setBarcode] = useState<string>("");
+
+  const handleEnterBarcode = () => {
+    if (!loading && data) {
+      const selectedPatient = data.patients.filter(
+        (patient) => patient.barcodeValue.toString() === barcode
+      );
+      if (selectedPatient.length > 0) {
+        // Found patient
+        const {
+          collectionPointId: { id: patientCCPId },
+          id,
+        } = selectedPatient[0];
+        // Redirect to patient profile
+        history.replace(`/patients/edit/${patientCCPId}/${id}`);
+      } else {
+        // No existing patient
+        history.replace(`/patients/new/${ccpID}`);
```

Comment:
How are we passing the barcode value to the new screen? The entered barcode value is not appearing. This works for the edit screen

Reply:
this is feature breaking, good catch @aaronabraham311